src/java.base/linux/classes/jdk/internal/platform/cgroupv1/CgroupV1Subsystem.java

```diff
@@ -0,0 +1,490 @@
+         * Host example:
+         * 5:memory:/user.slice
+         *
+         * Construct a path to the process specific memory and cpuset
+         * cgroup directory.
+         *
+         * For a container running under Docker from memory example above
+         * the paths would be:
+         *
+         * /sys/fs/cgroup/memory
+         *
+         * For a Host from memory example above the path would be:
+         *
+         * /sys/fs/cgroup/memory/user.slice
+         *
+         */
+        try (Stream<String> lines =
+                CgroupUtil.readFilePrivileged(Paths.get("/proc/self/cgroup"))) {
+
+            lines.map(line -> line.split(":"))
+                 .filter(line -> (line.length >= 3))
+                 .forEach(line -> setSubSystemControllerPath(subsystem, line));
+
+        } catch (IOException e) {
+            return null;
+        }
+
+        // Return Metrics object if we found any subsystems.
+        if (subsystem.activeSubSystems()) {
+            return subsystem;
+        }
+
+        return null;
+    }
+
+    /**
+     * createSubSystem objects and initialize mount points
+     */
+    private static void createSubSystemController(CgroupV1Subsystem subsystem, String[] mountentry) {
+        if (mountentry.length < 5) return;
+
+        Path p = Paths.get(mountentry[4]);
+        String[] subsystemNames = p.getFileName().toString().split(",");
+
+        for (String subsystemName: subsystemNames) {
+            switch (subsystemName) {
+                case "memory":
+                    subsystem.setMemorySubSystem(new CgroupV1MemorySubSystemController(mountentry[3], mountentry[4]));
+                    break;
+                case "cpuset":
+                    subsystem.setCpuSetController(new CgroupV1SubsystemController(mountentry[3], mountentry[4]));
+                    break;
+                case "cpuacct":
+                    subsystem.setCpuAcctController(new CgroupV1SubsystemController(mountentry[3], mountentry[4]));
+                    break;
+                case "cpu":
+                    subsystem.setCpuController(new CgroupV1SubsystemController(mountentry[3], mountentry[4]));
+                    break;
+                case "blkio":
+                    subsystem.setBlkIOController(new CgroupV1SubsystemController(mountentry[3], mountentry[4]));
+                    break;
+                default:
+                    // Ignore subsystems that we don't support
+                    break;
+            }
+        }
+    }
+
+    /**
+     * setSubSystemPath based on the contents of /proc/self/cgroup
+     */
+    private static void setSubSystemControllerPath(CgroupV1Subsystem subsystem, String[] entry) {
+        String controllerName;
+        String base;
+        CgroupV1SubsystemController controller = null;
+        CgroupV1SubsystemController controller2 = null;
+
+        controllerName = entry[1];
+        base = entry[2];
+        if (controllerName != null && base != null) {
```

Comment:
We are missing the fix of https://bugs.openjdk.java.net/browse/JDK-8254854 here. Old code was:
https://github.com/openjdk/jdk11u-dev/blob/194b66f6a72437345c458e747e24beb46352537a/src/java.base/linux/classes/jdk/internal/platform/cgroupv1/Metrics.java#L200..L202

See head commit at the time:
https://github.com/openjdk/jdk/commit/a0b687bf